core/src/main/java/com/google/instrumentation/trace/Tracer.java

```diff
@@ -60,25 +73,39 @@
 public final class Tracer {
   private static final Logger logger = Logger.getLogger(Tracer.class.getName());
   private static final Tracer INSTANCE =
-      new Tracer(
-          loadContextSpanHandler(Provider.getCorrectClassLoader(ContextSpanHandler.class)),
-          loadSpanFactory(Provider.getCorrectClassLoader(SpanFactory.class)));
-  private final ContextSpanHandler contextSpanHandler;
+      new Tracer(loadSpanFactory(Provider.getCorrectClassLoader(SpanFactory.class)));
   private final SpanFactory spanFactory;
 
+  @VisibleForTesting
+  Tracer(SpanFactory spanFactory) {
+    this.spanFactory = checkNotNull(spanFactory, "spanFactory");
+  }
+
   /**
-   * Returns the {@link Tracer} with the provided implementations for {@link ContextSpanHandler} and
-   * {@link SpanFactory}.
-   *
-   * <p>If no implementation is provided for any of the {@link Tracer} modules then no-op
-   * implementations will be used.
+   * Returns the {@link Tracer}. If no implementation is provided for any of the {@code Tracer}
+   * modules then no-op implementations will be used.
    *
-   * @return the {@link Tracer}.
+   * @return the {@code Tracer}.
    */
   public static Tracer getTracer() {
     return INSTANCE;
   }
 
+  // Any provider that may be used for SpanFactory can be added here.
+  @VisibleForTesting
+  static SpanFactory loadSpanFactory(ClassLoader classLoader) {
+    try {
+      // Because of shading tools we must call Class.forName with the literal string name of the
+      // class.
+      return Provider.createInstance(
+          Class.forName("com.google.instrumentation.trace.SpanFactoryImpl", true, classLoader),
+          SpanFactory.class);
+    } catch (ClassNotFoundException e) {
+      logger.log(Level.FINE, "Using default implementation for SpanFactory.", e);
```

Comment:
Yes. The expected behavior is that all the libraries like gRPC, google client APIs for cloud, etc will depend only on the API and the service owner will link the implementation.

If the service owner does not want tracing I think this should be fine, also all the tests for the libraries will print a warning message because they do not depend on the implementation.